gtsfm/ui/registry.py

```diff
@@ -0,0 +1,46 @@
+"""
+Setup for registry design pattern, which will save all classes that subclass
+GTSFMProcess for UI to display.
+
+Heavy inspiration from:
+https://charlesreid1.github.io/python-patterns-the-registry.html
+
+Author: Kevin Fu
+"""
+
+import abc
+from typing import Tuple, Dict
+
+
+class RegistryHolder(type):
+    """
+    Class that defines central registry and automatically registers classes
```

Comment:
nit: let's put this on one line, on the same line as the triple quote in line 16